packages/node/src/compartment/import-hook.js

```diff
@@ -16,27 +17,39 @@ const { freeze, keys, assign } = Object
  */
 
 /**
- * Import hook for exit modules, which are typically Node.js builtins in this
- * case.
+ * Given some `ns`, create a {@link ThirdPartyStaticModuleInterface}
  *
- * @type {ExitModuleImportHook}
- * @internal
+ * @param {unknown} ns Namespace
+ * @returns {Readonly<ThirdPartyStaticModuleInterface>}
  */
-export const importHook = async (specifier) => {
-  /** @type {object} */
-  const ns = await import(specifier)
+const makeStaticModuleInterface = (ns) => {
   return freeze(
     /** @type {ThirdPartyStaticModuleInterface} */ ({
+      // builtins have no imports (as far as we're concerned)
       imports: [],
-      exports: keys(ns),
+      exports: isObject(ns) ? keys(ns) : [],
       execute: (moduleExports) => {
+        // this is a no-op if `ns` is a non-object
         moduleExports.default = ns
         assign(moduleExports, ns)
       },
     })
   )
 }
 
+/**
+ * Import hook for exit modules, which are typically Node.js builtins
+ *
+ * @type {ExitModuleImportHook}
+ * @internal
+ */
+export const importHook = async (specifier) => {
+  /** @type {unknown} */
+  const ns = await import(specifier)
```

Comment:
(could) make sure specifier, if it starts with a protocol, is only allowed to proceed if the protocol is `node:` (and not `file:` or `http:` etc.)

Reply:
- [ ] put on list to do later